lib/merchant.rb

```diff
@@ -22,4 +22,19 @@ def invoices
     @invoices ||= @repository.find_invoices(id)
   end
 
+  def revenue
+    successful_invoice_items =  successful_invoices.map { |i| i.invoice_items }.flatten
+    successful_invoice_items.reduce(0) do |revenue, invoice_item|
+      revenue + (invoice_item.quantity * invoice_item.unit_price)
+    end
+  end
+
+  private
+
+  def successful_invoices
+    successful_transactions ||= invoices.map { |i| i.transactions }.flatten
+    successful_transactions.select! { |t| t.result == 'success'}
```

Comment:
Space missing inside }.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.